src/utils/address/checkAddressPartValid.js

```diff
@@ -0,0 +1,14 @@
+// @flow strict
+
+import {
+  add0x,
+} from '.'
+
+export function checkAddressPartValid(address: string): boolean {
+  const addr0x: string = add0x(address)
+
+  const isAddressLowerCase: boolean = /^0x[0-9a-f]{1,40}$/.test(addr0x)
```

Comment:
Please move regex creation out of function, because they are quite heavy and there is no need to create them each time you run the function